### PR TITLE
syncthing: 1.14.0 -> 1.15.1

### PR DESCRIPTION
**Pull request by bbigras**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bbigras** [comment]
Result of `nixpkgs-review pr 118647` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>syncthing-gtk</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>syncthing-cli</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>qsyncthingtray</li>
    <li>syncthing</li>
    <li>syncthing-discovery</li>
    <li>syncthing-relay</li>
  </ul>
</details>

**Comment by peterhoeg** [comment]
The standalone `cli` utility has now been integrated, so that needs to be fixed too: https://github.com/syncthing/syncthing/issues/6566

**Comment by bbigras** [comment]
> The standalone `cli` utility has now been integrated, so that needs to be fixed too: [syncthing/syncthing#6566](https://github.com/syncthing/syncthing/issues/6566)

Does the `syncthing-cli` package need to be deprecated with a note saying to use `syncthing cli`? I might be wrong, but I'm guessing we can't split the cli part.

**Comment by peterhoeg** [comment]
>   Does the syncthing-cli package need to be deprecated with a note saying to

Yep, exactly.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118647` at 97a29e86 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>syncthing-gtk</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/a974fed26690c876aa12b55456d61bf4">syncthing-cli</a></li>
  </ul>
</details>
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>qsyncthingtray</li>
    <li>syncthing</li>
    <li>syncthing-discovery</li>
    <li>syncthing-relay</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/networking/syncthing/default.nix:21:9:
```
   |
21 |         ./add-stcli-target.patch
   |         ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by bbigras** [comment]
I made the changes. Is "syncthing-cli is now included in the syncthing package" good enough or should it say to use "syncthing cli" from the syncthing package?

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -728,6 +728,7 @@ mapAliases ({
   sup = throw "sup was deprecated on 2019-09-10: abandoned by upstream";
   swfdec = throw "swfdec has been removed as broken and unmaintained."; # added 2020-08-23
   swtpm-tpm2 = swtpm; # added 2021-02-26
+  syncthing-cli = throw "syncthing-cli is now included in the syncthing package"; # added 2021-04-06
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:731]: Can we point this to to syncthing?

File: pkgs/applications/networking/syncthing/default.nix [hunk 2]
```diff
@@ -3,23 +3,20 @@
 let
   common = { stname, target, postInstall ? "" }:
     buildGoModule rec {
-      version = "1.14.0";
+      version = "1.15.1";
       name = "${stname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/syncthing/default.nix:7]:
```suggestion
      pname = stname;
      version = "1.15.1";
```

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (bbigras) [reply pkgs/top-level/aliases.nix]: Done

**Comment by bbigras** [comment]
I made the changes @SuperSandro2000

**Comment by risicle** [comment]
This is presumably to resolve https://nvd.nist.gov/vuln/detail/CVE-2021-21404

**Comment by risicle** [comment]
Also since this is golang I'm guessing this negative length will merely result in a panic, and thus be a DoS (despite github scoring it `7.5 HIGH`).

**Comment by risicle** [comment]
`nixpkgs-review` happy for me, linux x86_64.

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy for me, macos 10.14

**Pull request merged**